### PR TITLE
Resolve spec violations in WebSocket

### PR DESCRIPTION
**Pull request by riyafa**
>This PR resolves as much as possible of the spec violations. Not all spec violations have been resolved because they have not been resolved in netty itself.

>Following are added in this PR:

- Adds compression support for WebSocket Server

- Fixes issue with Reserved Bits usage on the server side

    Resolve ballerina-platform/ballerina-lang#12694

- Adds UTF-8 validation check for WebSocket frames.

    Resolve ballerina-platform/ballerina-lang#12696
    Resolve ballerina-platform/ballerina-lang#12695

- Fixes compression related issues

    Related issue: ballerina-platform/ballerina-lang#12698
    Related issue: ballerina-platform/ballerina-lang#12702
    The compression related issues are not fully fixed as they are issues in netty itself.

### CODE REVIEW

**Comment by riyafa** [comment]
Need to decide if we are going to support compression extension by default or if we are going to have it configurable.

File: components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/websocket/server/WebSocketServerHandshakeFunctionalityTestCase.java [hunk 1]
```diff
@@ -104,6 +104,57 @@ public void testUnsupportedWebSocketVersion() throws UnirestException {
...
+        urlConn.setRequestMethod("GET");
+        urlConn.setRequestProperty("Connection", "Upgrade");
+        urlConn.setRequestProperty("Upgrade", "websocket");
+        urlConn.setRequestProperty("Sec-WebSocket-Key", "dGhlIHNhbXBsZSBub25jZQ==");
+        urlConn.setRequestProperty("Sec-WebSocket-Version", "13");
+
+        String deflateHeader = "permessage-deflate";
+        urlConn.setRequestProperty("Sec-WebSocket-Extensions", deflateHeader);
+        urlConn.setRequestProperty("x-handshake", "true");
+
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/test/java/org/wso2/transport/http/netty/websocket/server/WebSocketServerHandshakeFunctionalityTestCase.java]: Extra white space.

**Comment by shafreenAnfar** [comment]
Question, Compression is another form extensions right ?

**Comment by riyafa** [comment]
Compression is an standard extension supported by WebSocket https://stackoverflow.com/a/19300336/3599535

**Comment by riyafa** [comment]
I am closing this PR till the compression extension is made configurable.

**Comment by riyafa** [comment]
compression is enabled by default and not configurable

**Pull request merged**